src/custom/components/AccountDetails/Transaction.tsx

```diff
@@ -63,22 +83,108 @@ const TransactionState = styled(OldTransactionState).attrs(
   ${(props): string | false => !!props.disableMouseActions && `pointer-events: none; cursor: none;`}
 `
 
+type CancellationModalProps = {
+  onDismiss: () => void
+  isOpen: boolean
+  orderId: string
+  summary: string | undefined
+}
+
+function CancellationModal(props: CancellationModalProps): JSX.Element | null {
+  const { orderId, isOpen, onDismiss, summary } = props
+  const shortId = shortenOrderId(orderId)
+
+  const [status, setStatus] = useState<'not started' | 'waiting for wallet' | 'error'>('not started')
+  const [error, setError] = useState('')
+  const [showMore, setShowMore] = useState(false)
+  const cancelOrder = useCancelOrder()
+
+  const toggleShowMore = () => setShowMore(showMore => !showMore)
+
+  useEffect(() => {
+    // Reset status every time orderId changes to avoid race conditions
+    setStatus('not started')
+  }, [orderId])
+
+  const onClick = useCallback(() => {
+    setStatus('waiting for wallet')
+    setError('')
+    cancelOrder(orderId)
+      .then(onDismiss)
+      .catch(e => {
+        setError(e.message)
+        setStatus('error')
+      })
+  }, [cancelOrder, onDismiss, orderId])
+
+  return (
+    <Modal isOpen={isOpen} onDismiss={onDismiss}>
+      {status === 'waiting for wallet' ? (
+        <ConfirmationPendingContent
+          onDismiss={onDismiss}
+          pendingText={`Soft cancelling order with id ${shortId}\n\n${summary}`}
+        />
+      ) : status === 'error' ? (
+        <TransactionErrorContent onDismiss={onDismiss} message={error || 'Failed to cancel order'} />
+      ) : (
+        <>
+          <ConfirmationModalContent
+            title="Cancel order"
+            onDismiss={onDismiss}
+            topContent={() => (
+              <>
+                <p>
+                  Are you sure you want to cancel the order <em>{shortId}</em>?
+                </p>
+                <p>{summary}</p>
+                <p>
+                  Keep in mind this is a soft cancellation{' '}
+                  <LinkStyledButton onClick={toggleShowMore}>[{showMore ? '- less' : '+ more'}]</LinkStyledButton>
+                </p>
+                {showMore && (
+                  <>
+                    <p>It will be taken into account in a best effort basis.</p>
```

Comment:
mmm not clear what this means

Reply:
You mean the sentence or the logic?

For the sentence, means that you can (try to) cancel, but we don't guarantee it'll be cancelled.

Not good?

Reply:
I've removed this bit on https://github.com/gnosis/cowswap/pull/760